packages/fetch-router/src/lib/response-helpers/file.ts

```diff
@@ -0,0 +1,337 @@
+    }
+
+    // If-Range support: https://httpwg.org/specs/rfc9110.html#field.if-range
+    if (
+      headers.ifRange.matches({
+        etag,
+        lastModified,
+      })
+    ) {
+      if (!range.canSatisfy(fileToSend.size)) {
+        return new Response('Range Not Satisfiable', {
+          status: 416,
+          headers: new SuperHeaders({
+            contentRange: { unit: 'bytes', size: fileToSend.size },
+          }),
+        })
+      }
+
+      let normalizedRanges = range.normalize(fileToSend.size)
+
+      // We only support single ranges (not multipart)
+      if (normalizedRanges.length > 1) {
+        return new Response('Range Not Satisfiable', {
+          status: 416,
+          headers: new SuperHeaders({
+            contentRange: { unit: 'bytes', size: fileToSend.size },
+          }),
+        })
+      }
+
+      let { start, end } = normalizedRanges[0]
+      let { size } = fileToSend
+
+      return new Response(fileToSend.slice(start, end + 1), {
+        status: 206,
+        headers: new SuperHeaders(
+          omitNullableValues({
+            contentType,
+            contentLength: end - start + 1,
+            contentRange: { unit: 'bytes', start, end, size },
+            etag,
+            lastModified,
+            cacheControl,
+            acceptRanges,
+          }),
+        ),
+      })
+    }
+  }
+
+  return new Response(request.method === 'HEAD' ? null : fileToSend, {
+    status: 200,
+    headers: new SuperHeaders(
+      omitNullableValues({
+        contentType,
+        contentLength,
+        etag,
+        lastModified,
+        cacheControl,
+        acceptRanges,
+      }),
+    ),
+  })
+}
+
+function generateWeakETag(file: File): string {
+  return `W/"${file.size}-${file.lastModified}"`
+}
+
+type OmitNullableValues<T> = {
+  [K in keyof T as T[K] extends null | undefined ? never : K]: NonNullable<T[K]>
+}
+
+function omitNullableValues<T extends Record<string, any>>(headers: T): OmitNullableValues<T> {
+  let result: any = {}
+  for (let key in headers) {
+    if (headers[key] != null) {
+      result[key] = headers[key]
+    }
+  }
+  return result
+}
+
+/**
+ * Computes a digest (hash) for a file.
+ *
+ * @param file - The file to hash
+ * @param digestOption - Algorithm name or custom digest function
+ * @returns The computed digest as a hex string
+ */
+async function computeDigest(
+  file: File,
+  digestOption: DigestAlgorithm | FileDigestFunction,
+): Promise<string> {
+  return typeof digestOption === 'function'
+    ? // Custom digest function
+      await digestOption(file)
+    : // Use SubtleCrypto with algorithm name
+      await hashFile(file, digestOption)
+}
+
+/**
+ * Hashes a file using SubtleCrypto.
+ *
+ * @param file - The file to hash
+ * @param algorithm - Hash algorithm name (e.g., 'SHA-256')
+ * @returns The hash as a hex string
+ */
+async function hashFile(file: File, algorithm: string): Promise<string> {
+  let buffer = await file.arrayBuffer()
```

Comment:
This is totally just an optimization, but it's a shame we have to buffer the file here. Did you explore any way to create the digest incrementally, in a stream? I don't think `crypto.subtle` supports it, but I know `node:crypto` does. `node:crypto` is widely supported by all the various vendors we support. Maybe consider using it here?